pkg/api/v1/endpoint.go

```diff
@@ -92,44 +86,33 @@ func (e *Endpoint) DeepCopy() *Endpoint {
 	return &result
 }
 
-// SyncEndpoints adds the given list of endpoints in the internal endpoint
-// slice. All endpoints in the internal endpoint slice that are not in the given
-// 'newEps' slice will be marked as "deleted".
+// SyncEndpoints adds the given list of endpoints to the internal endpoint
+// slice.
 func (es *Endpoints) SyncEndpoints(newEps []*Endpoint) {
 	if len(newEps) == 0 {
 		return
 	}
 	es.mutex.Lock()
 	defer es.mutex.Unlock()
-	// Mark all endpoints not found as deleted
-	for _, ep := range es.eps {
-		if ep.Deleted != nil {
-			continue
-		}
-		found := false
-		for _, updatedEp := range newEps {
-			if ep.EqualsByID(updatedEp) {
-
-				found = true
-				break
-			}
-		}
-		// If we haven't found it, it means we have lost, or haven't receive
-		// yet, an event signalizing that this endpoint was deleted.
-		if !found {
-			t := time.Now()
-			// TODO: remove leftover endpoints if the timestamp of the last
-			//  flow written is after the endpoint was deleted.
-			//  This requires a method in the ring buffer that returns
-			//  the older flow written.
-			ep.Deleted = &t
-		}
-	}
-
 	// Add the endpoint to the list of endpoints.
 	for _, updatedEp := range newEps {
 		es.updateEndpoint(updatedEp)
 	}
+	// some endpoints were deleted, remove them
+	if len(es.eps) != len(newEps) {
```

Comment:
I don't think this if statement is correct, the length itself is not sufficient to detect deletion. If `es.eps=={1,2,3}` and `newEps=={1,4,5}`, this branch would never be taken even though we have two deleted endpoints (`2` and `3`) that must be removed via `es.deleteEndpoint`.

Reply:
The lines above actually add new endpoints. So once you reach line `102`, you'd actually have `es.eps == {1,2,3,4,5}` which is of a different size than `newEps == {1,4,5}`. So in fact, the branch would be taken.

Reply:
Ah! Yes, you are right, I missed that. Sorry for the noise :)